plugin.rb

```diff
@@ -26,7 +26,7 @@ def self.access_restricted(guardian, topic, user)
       hasBeenLocked = true
 
       # if topic is not restricted, dont lock it
-      if !topic.custom_fields["topic_restricted_access"]
+      if !topic.custom_fields["topic_restricted_access"].nil? && !topic.custom_fields["topic_restricted_access"]
```

Comment:
Why are you removing this? This commit needs more details. It looks like there is a purposeful check to see if the topic has the "topic_restricted_access field before accessing it to avoid whatever the nullpointer error in ruby is.

Reply:
That should of been an or. Fixing and adding to commit message

Reply:
There's errors in the site logs about the topics not having any custom field at this line, and if that is the  case access should be restricted

Reply:
Sorry, that comment was probably confusing. I mistook the added line for the removed line.

Reply:
Why should that be and or and not an and?

Reply:
well, it pointed out a glaring typo. I did end up removing a case instead of adding it. I'm just going to finish building this on my test site to run some checks before i merge it

Reply:
It should be || because if the topic has no custom fields we can assume it should be treated like a private topic and return false here to restrict access. that is on top of checking if we explicitly say to restrict access from having that custom field set to true

Reply:
seems to work without errors now. I'm gonna merge this

Reply:
Merged without approval?

Reply:
uh, you approved it 12 minutes ago